### PR TITLE
java-interop: add extra properties to GreyhoundConfig

### PR DESCRIPTION
**Pull request by b41somp**
- Modify GreyhoundConfig to accept extraProperties map parameter.
- Add Unit test case (`GreyhoundBuilderTest.configure_consumer_with_timeout_custom_props`) but it is still failing; seem like properties setting doesn't have any affect on kafka consumer/producer. Please help to verify this.

### CODE REVIEW

**Comment by natansil** [comment]
restarting PR to try and trigger build on _github actions_

**Comment by natansil** [comment]
@b41somp, great job so far.
regarding the failing test.
Maybe a better way to test would be to check with an additional consumer with a custom property that changes it's group name, but both consumers using the same handler.

and checking that invocations happened double.

for that you would have to go back to your original idea of having the custom props on consumer level - which is not such a bad idea, as there are props that are unique to consumers...

In terms of testing the producer, may I suggest to add another test that uses the `max.request.size ` with low value, and check the produce fails with `RecordTooLargeException`

WDYT?

**Comment by b41somp** [comment]
I have managed to add producer and consumer test cases to test kafka properties. 
For producer test case, I test on `max.request.size` property and on consumer I test on `max.poll.interval.ms` property with value '0'. It doesn't look good to me on consumer test case but I tried many failure ways except this one. Please help to verify.

File: java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java [hunk 1]
```diff
@@ -261,6 +260,73 @@ public void configure_consumer_with_exponential_blocking_retry_policy_max_interv
...
 
+    @Test
+    public void configure_producer_with_max_request_size_custom_props() throws Exception {
+        GreyhoundConfig config = new GreyhoundConfig(environment.kafka().bootstrapServers(), new HashMap<String, String>() {
+            {
+                put("max.request.size", "1");
+            }
+        });
+        GreyhoundProducerBuilder producerBuilder = new GreyhoundProducerBuilder(config);
+        GreyhoundConsumersBuilder consumersBuilder = new GreyhoundConsumersBuilder(config)
```
Comment (natansil) [inline java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java:271]: consumersBuilder seems redundant in this test case. can you please remove it?

File: java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java [hunk 2]
```diff
@@ -261,6 +260,73 @@ public void configure_consumer_with_exponential_blocking_retry_policy_max_interv
...
+            assertTrue(kafkaError.getCause() instanceof RecordTooLargeException);
+        }
+    }
+
+    @Test
+    public void configure_consumer_with_custom_props() throws Exception {
+        int numOfMessages = 1;
+        List<ConsumerRecord<Integer, String>> consumedRecords = new LinkedList<>();
+        CountDownLatch lock = new CountDownLatch(numOfMessages);
+        GreyhoundConfig producerConfig = new GreyhoundConfig(environment.kafka().bootstrapServers());
```
Comment (natansil) [inline java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java:301]: `producerConfig` seems redundant in this test

File: java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java [hunk 3]
```diff
@@ -261,6 +260,73 @@ public void configure_consumer_with_exponential_blocking_retry_policy_max_interv
...
+        GreyhoundConfig consumerConfig = new GreyhoundConfig(environment.kafka().bootstrapServers(), new HashMap<String, String>() {
+            {
+                put("max.poll.interval.ms", "0");
+            }
+        });
+        GreyhoundProducerBuilder producerBuilder = new GreyhoundProducerBuilder(producerConfig);
```
Comment (natansil) [inline java-interop/src/test/java/com/wixpress/dst/greyhound/java/GreyhoundBuilderTest.java:307]: `producerBuilder` seems redundant in this test case

**Review by natansil — COMMENTED** [review]
Amazing job @b41somp.
please see very small issue in comments.

**Comment by b41somp** [comment]
fixed

**Review by natansil — APPROVED** [review]

**Comment by natansil** [comment]
performing close/open to trigger sync to private

**Comment by natansil** [comment]
@b41somp the commits of these PR were sync to our private repo and synced back here with [this](https://github.com/wix/greyhound/pull/307) PR.
The PR is merged and I'm cutting a new version for Maven Central.

All of the team at Wix really appreciates your meaningful contribution to Greyhound!

**Pull request closed without merging**